app/src/main/java/com/example/stromtracker/MainActivity.kt

```diff
@@ -16,28 +17,49 @@ import androidx.navigation.ui.setupWithNavController
 import androidx.drawerlayout.widget.DrawerLayout
 import androidx.appcompat.app.AppCompatActivity
 import androidx.appcompat.widget.Toolbar
+import androidx.lifecycle.Observer
+import com.example.stromtracker.database.DataRepository
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.ProcessLifecycleOwner
+import androidx.lifecycle.ViewModelProviders
+import com.example.stromtracker.database.Geraete
+import com.example.stromtracker.database.Haushalt
+import com.example.stromtracker.ui.geraete.GeraeteViewModel
 
 class MainActivity : AppCompatActivity() {
 
     private lateinit var appBarConfiguration: AppBarConfiguration
+    private lateinit var sp:Spinner
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
         val toolbar: Toolbar = findViewById(R.id.toolbar)
+        var geraeteViewModel: GeraeteViewModel =  ViewModelProviders.of(this).get(GeraeteViewModel::class.java)
+        val haushaltItems:ArrayList<Haushalt> = ArrayList()
+
 
 
         setSupportActionBar(toolbar)
 
+
+
         val drawerLayout: DrawerLayout = findViewById(R.id.drawer_layout)
         val navView: NavigationView = findViewById(R.id.nav_view)
         val navController = findNavController(R.id.nav_host_fragment)
-        val items = arrayOf("Haushalt1", "Haushalt2", "Haushalt3", "Haushalt4")
 
-        val sp: Spinner = navView.menu.findItem(R.id.nav_haushalt).getActionView() as Spinner
-        val adapter = ArrayAdapter<String>(this, android.R.layout.simple_spinner_dropdown_item, items)
-        sp.adapter = adapter
+        geraeteViewModel.getAllHaushalt().observe(
+            this,
+            Observer { haushalte ->
+                if (haushalte != null) {
 
+                    haushaltItems.clear()
+                    haushaltItems.addAll(haushalte)
+                    sp= navView.menu.findItem(R.id.nav_haushalt).actionView as Spinner
+                    val adapter = ArrayAdapter<Haushalt>(this, android.R.layout.simple_spinner_dropdown_item, haushaltItems)
+                    sp.adapter = adapter
+                }
+            })
```

Comment:
Damit haben wir jetzt den Spinner und befüllen diesen jetzt mit den Daten aus der Datenbank oder?

Was wir noch machen müssten, ist dann in neine globale Variable vlt. den ajtuell ausgewählten haushalt zu schreiben um unnötige Datenbankzugriffe in den Fragmnents zu verhindern. Wäre das sinnvoll?

Reply:
Der Tobias hatte das doch so gelöst, dass wir uns den Spinner in die jeweiligen Fragments holen und durch den dann den ausgewählten Haushalt kriegen.

Extra Datenbankzugriffe brauchen wir dafür nicht, das wird ja durch LiveData automatisch geupdatet, aber nur in der MainActivity.

Reply:
Ah ok. Verstehe. Wenn dann ein neuer Haushalt erzeugt wird, kommt der also dann auch automatisch in den Spinner?

Reply:
Ja, das wird ja durch den Observer automatisch gesetzt.